encord/objects/internal_helpers.py

```diff
@@ -549,6 +549,7 @@ def _infer_attribute_from_answer(
                 "Cannot infer the attribute if a list of answers is empty. Please provide the " "attribute explicitly."
             )
 
+        assert isinstance(answer[0], Option)  # Narrowing type here as sequence can contain only Options
```

Comment:
here we seem to be confident, however if we're less than 100% confident I want to say that if clients are running with optimised Python, i.e. `-O` flag, then[ these asserts will be ignored](https://discuss.python.org/t/stop-ignoring-asserts-when-running-in-optimized-mode/13132) (if it hasn't changed so far). 
So an `if ... raise` is in that case safer.

Reply:
100% agree, currently we're using asserts only to hint mypy, and I think we can keep it as a code convention.
In this case the problem is that we check isinstance(answer, Sequence) above, as checking for isinstance(answer, Sequence[Option]) is impossible due to the python limitations.
So in my opinion it is safe here, and we don't really need a real runtime check. 

But I agree that it is still fishy, and perhaps we will be able to refactor it out in the future